libraries/script-engine/src/XMLHttpRequestClass.cpp

```diff
@@ -117,7 +122,7 @@ void XMLHttpRequestClass::setReadyState(ReadyState readyState) {
     if (readyState != _readyState) {
         _readyState = readyState;
         if (_onReadyStateChange.isFunction()) {
-            _onReadyStateChange.call(QScriptValue::NullValue);
+            _onReadyStateChange.call(_onReadyStateChange.engine()->nullValue());
```

Comment:
Unsure about this, although it might be splitting hairs.  The only time this would be different is if onReadyStateChange somehow came from a different thread than the XMLHttpRequest object did.  Which... really shouldn't happen... but I have a _slight_ preference to the existing code.

Reply:
I see what you mean...in that case, shouldn't the other ones in this file be using the appropriate _responseData/_onTimeout engines then?